### PR TITLE
Feral balance audit

### PR DESCRIPTION
**Pull request by Karol1223**
#### Summary
Balance "Make ferals actually feel like human enemies instead of weaker zombies with range attacks"

#### Purpose of change
Ferals have a history of balance discussions regarding their ranged attacks, but the truth is their melee is what's really bad currently. A pipe feral is for all intents and purposes weaker than a baseline zombie - the only upsides it has is the higher speed and a ranged attack, but it's melee (when using a weapon!) is incomparably worse than a walking corpse. Yet the weapon it uses isn't bad. So what went wrong?

#### Describe the solution
- Rebalance feral health to be that of an average str 8 character's torso health (84 health, for some this was a buff but for some this was a nerf)
- Rebalance feral base melee to a single d6, the apparent melee damage for a 3 melee skill unarmed hit from a str 8 player
- Rebalance feral rock throw to use str 8 instead of 7, but have one lower throwing skill
- All ferals now have anger triggers for ``PLAYER_CLOSE`` and ``PLAYER_WEAK``. Realistically this should not change anything for 90% of them, but it allowed me to cut down on some JSON lines when it comes to using ``extend``
- All ferals now have their special pathing assigned, and they all have a minimum of 45 day vision (this is 3/4 of player's day vision, still, but setting it at 60 would make them insanely lethal)
- Add new special attacks with weapon stats from weapons the specific feral is using (only applies to melee, ranged weapons already kinda do that and I don't like dabbling in that)
- Removed the unnecessary night vision buff for the crazed survivalist and mad militia - useless for a monster with such high luminance
- Lowered mad militia's speed to 100 because with it being this armored up it shouldn't be this fast, given the treatment some other armored-up ferals got
- Slightly lowered the night vision of the feral marine
- Armored zombies are now slower (speed 100 -> 85) and no longer train synthetic armors proficiency
- Used ``copy-from``, ``extend`` and ``delete`` when applicable to shorten the file length

**Feral Checklist**:
- [x] feral human
- [x] feral cop
- [x] feral marine
- [x] feral biker
- [x] feral servant
- [x] feral officer
- [x] feral prepper
- [x] feral crewman
- [x] feral soldier
- [x] feral swimmer
- [x] armored feral
- [x] cunning feral
- [x] feral mechanic
- [x] mad militia
- [x] well-dressed feral
- [x] feral archeologist
- [x] deranged axeman
- [x] feral security guard
- [x] crazed survivalist
- [x] mad scientist
- [x] fungal ferals
- [x] starers

#### Describe alternatives you've considered
1. I have no idea if ``accuracy`` is a good metric to translate ``to-hit`` from weapons to. If it is not please let me know and I'll be sad then figure something out
2. Ideally we'd have some sort of disarming but that's beyond my scope of skills
3. I briefly considered adding some sort of "maimed arms" effect from weakpoint hits like Venera suggested but I am not sure how to do that. Maybe later
4. Changing the speed to always be 100, as some like the feral cop use 95 for example. All in all, I don't know exactly how speed for the player is calculated so I wouldn't know when and how much the gear would slow them down

#### Testing
Tested it on the pipe feral and it looked to be working as intended. When it got into melee it started beating the shit out of me with a pipe.

### CODE REVIEW

**Comment by snipercup** [comment]
Considering the comments in #58645 my suggestion would be to lower the throwing skill for ferals, but not increase their strength. If you increase their strength, they will have a greater throwing range and have an even greater advantage when they attack from behind a wall of zombies. Having a greater throwing range also makes them a threat to archers, who can currently only barely out-range them (which is weird since an arrow should fly further then a thrown rock).

**Comment by Karol1223** [comment]
Are you sure that's necessary? Can't ranges be defined manually? I'll have to look more into how this works. I wanted it to be str 8 so we don't have discrepancies between their stats but if it's that big of a deal I may have to lower the throw strength back down like you said.

**Comment by fairyarmadillo** [comment]
Feral divers are completely ridiculous and should probably be removed. Diving equipment is very difficult to use and of limited utility (just look at how many people accidentally die trying to SCUBA dive every year) but somehow these insane people are doing it? And just like, randomly out in the rivers, all over the place?

**Comment by Karol1223** [comment]
> Feral divers are completely ridiculous and should probably be removed. Diving equipment is very difficult to use and of limited utility (just look at how many people accidentally die trying to SCUBA dive every year) but somehow these insane people are doing it? And just like, randomly out in the rivers, all over the place?

I'm not sure I want to touch what ferals fit the lore and what don't because:
1. Divers and archeologists would have to go
2. ANYTHING that uses firearms, in my honest to god opinion, would have to go as well. Operating firearms may not be *that* hard to learn (though I wouldn't know, European here) but the amount of arguments that ended with "a feral with grenades would have blown them up before they met the player" leads me to believe that the exact same thought process should be applied to ammo for firearms myself.
3. I'd have to retouch Starers, likely, as I've had my eye on them for a longer while now and they make limited amount of sense within our lore. They're salvageable but they need some edits.

All in all if I cared in this specific PR it'd inflate the size out of proportions so I think I'll just rebalance them all right now and possibly murder any later down the line.

**Comment by Karol1223** [comment]
> Regardless, I'm not touching any of that here.

He said, lying as easily as he breathed. Anyway, #67707 handles that. This PR will wait until that's merged so I don't have to do work for something that won't survive long term.

**Comment by Karol1223** [comment]
Well ain't that just swell

**Comment by Maleclypse** [comment]
I’m going to test out these changes a bit

File: data/json/monster_special_attacks/feral_weapon_attacks.json [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+  },
+  {
+    "type": "monster_attack",
+    "attack_type": "melee",
+    "id": "feral_weapon_knife_chef",
+    "cooldown": 1,
+    "move_cost": 72,
+    "accuracy": 2,
+    "//": "Feral melee skill + weapon's to hit, so 3-1=2",
+    "damage_max_instance": [ { "damage_type": "bash", "amount": 2 }, { "damage_type": "pierce", "amount": 16 } ],
```
Comment (Maleclypse) [inline data/json/monster_special_attacks/feral_weapon_attacks.json:122]:
```suggestion
    "damage_max_instance": [ { "damage_type": "bash", "amount": 2 }, { "damage_type": "stab", "amount": 16 } ],
```

File: data/json/monster_special_attacks/feral_weapon_attacks.json [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+  },
+  {
+    "type": "monster_attack",
+    "attack_type": "melee",
+    "id": "feral_weapon_rapier",
+    "cooldown": 1,
+    "move_cost": 107,
+    "accuracy": 6,
+    "//": "Feral melee skill + weapon's to hit, so 4+2=6",
+    "damage_max_instance": [ { "damage_type": "bash", "amount": 3 }, { "damage_type": "pierce", "amount": 29 } ],
```
Comment (Maleclypse) [inline data/json/monster_special_attacks/feral_weapon_attacks.json:154]:
```suggestion
    "damage_max_instance": [ { "damage_type": "bash", "amount": 3 }, { "damage_type": "stab", "amount": 29 } ],
```

File: data/json/monster_special_attacks/feral_weapon_attacks.json [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+  },
+  {
+    "type": "monster_attack",
+    "attack_type": "melee",
+    "id": "feral_weapon_knife_combat",
+    "cooldown": 1,
+    "move_cost": 82,
+    "accuracy": 5,
+    "//": "Feral melee skill + weapon's to hit, so 6-1=5",
+    "damage_max_instance": [ { "damage_type": "bash", "amount": 4 }, { "damage_type": "pierce", "amount": 26 } ],
```
Comment (Maleclypse) [inline data/json/monster_special_attacks/feral_weapon_attacks.json:282]:
```suggestion
    "damage_max_instance": [ { "damage_type": "bash", "amount": 4 }, { "damage_type": "stab", "amount": 26 } ],
```

File: data/json/monster_special_attacks/feral_weapon_attacks.json [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+  {
+    "type": "monster_attack",
+    "attack_type": "melee",
+    "id": "feral_weapon_spear_knife",
+    "cooldown": 1,
+    "move_cost": 119,
+    "accuracy": 2,
+    "range": 2,
+    "//": "Feral melee skill + weapon's to hit, so 3-1=2",
+    "damage_max_instance": [ { "damage_type": "bash", "amount": 4 }, { "damage_type": "pierce", "amount": 26 } ],
```
Comment (Maleclypse) [inline data/json/monster_special_attacks/feral_weapon_attacks.json:299]:
```suggestion
    "damage_max_instance": [ { "damage_type": "bash", "amount": 4 }, { "damage_type": "stab", "amount": 26 } ],
```

File: data/json/monster_special_attacks/feral_weapon_attacks.json [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+  {
+    "type": "monster_attack",
+    "attack_type": "melee",
+    "id": "feral_weapon_feral_m4_carbine_bayonet",
+    "cooldown": 1,
+    "move_cost": 196,
+    "accuracy": 3,
+    "range": 2,
+    "//1": "Feral melee skill + weapon's to hit, so 4-1=3",
+    "damage_max_instance": [ { "damage_type": "bash", "amount": 12 }, { "damage_type": "pierce", "amount": 26 } ],
```
Comment (Maleclypse) [inline data/json/monster_special_attacks/feral_weapon_attacks.json:445]:
```suggestion
    "damage_max_instance": [ { "damage_type": "bash", "amount": 12 }, { "damage_type": "stab", "amount": 26 } ],
```

**Review by Maleclypse — COMMENTED** [review]
Errors popped up during testing.

**Comment by Karol1223** [comment]
Thanks for catching those! Still confused why the tests didn't catch them, though... or why I didn't see those during my testing. But oh well.

**Comment by Maleclypse** [comment]
Very deadly.  Their spawns will likely need to be reduced but it will take more playtesters than me to be sure of that.

**Comment by Karol1223** [comment]
> Very deadly.  Their spawns will likely need to be reduced but it will take more playtesters than me to be sure of that.  

So sounds like its working as intended then! I don't know if reducing spawns is that necessary, because the more lethal ones are either location exclusive or rare as is (I have seen about 5 total axemen in my 3 year save) and the common ones with the pipe and crowbar shouldn't be at the tipping point of destroying balance. 

Also it only occured to me now I forgot to adjust the swing speed for the pipe feral...

**Comment by zachary-kaelan** [comment]
Are the ferals still packing Armor Piercing Fin Stabilized Discarding Sabot Rocks?

**Comment by Karol1223** [comment]
> Are the ferals still packing Armor Piercing Fin Stabilized Discarding Sabot Rocks?

I have not audited the rock throw much. As far as I'm concerned they're using the same rock throw that the player would be. And if the player gets to do that, they should get to do that too.

**Comment by fairyarmadillo** [comment]
> Are the ferals still packing Armor Piercing Fin Stabilized Discarding Sabot Rocks?

They have never had these. Their rocks do about 5-12 bash damage which is usually gonna be less than what they hit for in melee. Wear armor or stand further than five tiles from them, or get to melee.

**Comment by fairyarmadillo** [comment]
> EDIT: or maybe not? They're using a fake gun. Oh well, not my problem. I didn't touch that enough to bother.

The fake gun functions about like a player throw at equivalent skill.

**Pull request merged**